JoinColony/colonyCDapp
# Cancelling non-staked expenditure as non-owner (using motions)

## Description

This PR introduces new saga, `cancelExpenditureMotion`, that allows creating a motion to cancel an expenditure as a non-owner.

Related block-ingestor PR: https://github.com/JoinColony/block-ingestor/pull/191

## Testing

1. Install and enable voting reputation extension.
2. Create an expenditure.
3. At any point of the flow (before finalization), create a motion to cancel using the testing UI.
4. Check the expenditure in the DB. Under `motions`, there should be a new motion linked to an action of type `CANCEL_EXPENDITURE_MOTION`. From the action, it should also be possible to access the linked expenditure:
```gql
query ListExpenditures {
  listExpenditures(filter: {nativeId: {ge: 33}}) {
    items {
      nativeId
      status
      motions {
        items {
          id
          action {
            type
            expenditure {
              nativeId
            }
          }
        }
      }
    }
  }
}
```
5. Progress the motion to finalization. Upon finalizing, the status of the expenditure should change to cancelled.

Resolves #1998
